src/stardict_lib.cpp

```diff
@@ -573,7 +574,9 @@ bool OffsetIndex::load_cache(const std::string &url)
             continue;
         if (strncmp(mf.begin(), CACHE_MAGIC, strlen(CACHE_MAGIC)) != 0)
             continue;
-        memcpy(&wordoffset[0], mf.begin() + strlen(CACHE_MAGIC), wordoffset.size() * sizeof(wordoffset[0]));
+        if (*(guint32 *)(mf.begin() + strlen(CACHE_MAGIC)) != CACHE_MAGIC_BYTES)
```

Comment:
For some CPU it is bad to access 32bit word not via 32bit aligned pointer,
it would be better do something like:
```
guint32 tmp;
memcpy(&tmp, mf.begin() + strlen(CACHE_MAGIC), sizeof(tmp));
if (tmp != CACHE_MAGIC_BYTES) {
```

Reply:
Good point, patch updated.